sql: RSG needs to be updated to keep track of read-only transaction state

https://teamcity.cockroachdb.com/viewLog.html?buildId=436128&buildTypeId=Cockroach_Nightlies_RandomSyntaxTests&tab=buildLog&_focus=2120

```
panic: pq: cannot execute CREATE DATABASE in a read-only transaction
```

--- a/pkg/sql/tests/rsg_test.go
+++ b/pkg/sql/tests/rsg_test.go
@@ -106,6 +106,9 @@ func TestRandomSyntaxGeneration(t *testing.T) {
 		if strings.HasPrefix(s, "SET SESSION CHARACTERISTICS AS TRANSACTION") {
 			return errors.New("setting session characteristics is unsupported")
 		}
+		if strings.Contains(s, "READ ONLY") || strings.Contains(s, "read_only") {
+			return errors.New("READ ONLY settings are unsupported")
+		}
 		// Recreate the database on every run in case it was dropped or renamed in
 		// a previous run. Should always succeed.
 		if err := db.exec(`CREATE DATABASE IF NOT EXISTS ident`); err != nil {

